src/find-configuration-file.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+import fs from "fs";
+import path from "path";
+
+import {ancesdirOrCurrentDir} from "./ancesdir-or-currentdir.js";
+
+import type {ILog} from "./types.js";
+
+export const checkSyncRcNames = [".checksyncrc", ".checksyncrc.json"];
+
+export default function findConfigurationFile(
+    rootMarker: ?string,
+    log: ILog,
+): ?string {
+    log.verbose(() => `Looking for configuration file based on root marker...`);
+
+    // Let's look for the root marker relative to our current working directory
+    // and see if there's one adjacent to it.
+    try {
+        const rootPath = ancesdirOrCurrentDir(process.cwd(), rootMarker);
+        for (const rcPath of checkSyncRcNames.map((rc) =>
+            path.join(rootPath, rc),
+        )) {
+            log.verbose(() => `Checking ${rcPath}...`);
+            if (fs.existsSync(rcPath)) {
+                log.verbose(() => `Found configuration file: ${rcPath}`);
+                return rcPath;
+            }
+        }
+    } catch (e) {
+        // Root marker not found based on the current working directory.
+        // We'll just ignore this.
+        log.verbose(() => `Root marker based configuration not found`);
+    }
+
+    log.verbose(
+        () =>
+            `Looking for configuration based on current working directory: ${process.cwd()}`,
+    );
```

Comment:
It won't appear after that since that found message will return right away.

Or do you mean the way the code reads? I'm not sure there's a nice way of organizing it so that the messages appear in the logical order they'll be output.

Reply:
My comment was about what appears to users.  I didn't notice the short-circuit.